src/autotype/AutoTypeSelectDialog.cpp

```diff
@@ -108,6 +111,23 @@ void AutoTypeSelectDialog::setMatches(const QList<AutoTypeMatch>& matches, const
     }
 }
 
+void AutoTypeSelectDialog::setUrl(const QString& url)
+{
+    QString host;
+    if (url.startsWith("file://")) {
+        host = url;
+    } else {
+#ifdef WITH_XC_BROWSER
+        host = browserService()->getBaseDomainFromUrl(url);
```

Comment:
At first I moved the whole `getBaseDomainFromUrl` under Tools, but noticed that the function requires QHostAddress which needs QtNetwork. Thus, we are calling it now from BrowserService. Another solution would be to replace the whole IP address identification with e.g. Regex so this could be moved to Tools.

Reply:
I'll evaluate but this creates an interesting dependency. I'd rather the browser service pass in the desired search on its own. Might need a refactor of my code.

Reply:
`getBaseDomainFromUrl()` has now been moved to Tools without QtNetwork dependency.